qcelemental/molutil/connectivity.py

```diff
@@ -16,18 +16,18 @@ def guess_connectivity(
 
     Parameters
     ----------
-    symbols : np.ndarray
+    symbols
         The molecular symbols (e.g., 'Zr', 'C')
-    geometry : np.ndarray
+    geometry
         The molecular geometry in Bohr
-    threshold : float, optional
+    threshold
         Tunes the covalent radii metric safety factor.
-    default_connectivity : Optional[float], optional
+    default_connectivity
         Provides a default connectivity value
 
     Returns
     -------
-    List[Union[Tuple[int, int], Tuple[int, int, float]]]
+    ~typing.List[~typing.Union[~typing.Tuple[int, int], ~typing.Tuple[int, int, float]]]
```

Comment:
Do we need `~typing...` here? They were stripped out of other modules so I want to make sure we're not adding extra bulk.

Reply:
I can answer that with your `path : Union[str, Path] -> path` query.

So with newer sphinx auto/numpy/etc extensions, sphinx can entirely use typing info for parameters from the function signature. having it also in the docstring only overrides the signature typing and changes up the formatting for the worse. so no need for it in docstrings anymore.

If you put the return typing in the signature and don't need a comment for it, the autodocs can use the signature return type. If you do want a docstring comment for the return, the return type does have to be repeated in the docstring to be compatible with numpy style. you can have plain `Union[str, Tuple]` there, but the Union and Tuple won't be linked back to python docs as they normally would. `typing` allows them to be a linked and `~` suppresses the typing.